src/app/metric/resource.ts

```diff
@@ -0,0 +1,32 @@
+import Metric from 'src/models/metric';
+import {
+  ICreateMetricPayload,
+  IMetricDocument,
+  IUpdateMetricPayload
+} from 'src/types';
+import * as Transformer from './transformer';
+
+export const list = async (
+  query: { meta?: Record<string, string> } = {}
+): Promise<IMetricDocument[]> => {
+  return Metric.list(query);
+};
+
+export const create = async (
+  payload: ICreateMetricPayload
+): Promise<IMetricDocument> => {
+  const data = await Transformer.create(payload);
+  const metric = await Metric.create(data);
```

Comment:
you can remove await and return directly like others
`return Metric.create(data);`